core/src/main/java/com/xmartlabs/bigbang/core/controller/CoreSessionController.java

```diff
@@ -0,0 +1,70 @@
+package com.xmartlabs.bigbang.core.controller;
+
+import android.content.SharedPreferences;
+import android.support.annotation.CheckResult;
+import android.support.annotation.NonNull;
+
+import com.annimon.stream.Optional;
+import com.xmartlabs.bigbang.core.model.SessionType;
+
+import javax.inject.Inject;
+
+import io.reactivex.Single;
+import lombok.RequiredArgsConstructor;
+
+/**
+ * Controller that manages the Session of the Application.
+ *
+ * The Session will be stored via the {@link SharedPreferencesController}.
+ */
+@RequiredArgsConstructor
+public class CoreSessionController extends Controller {
+  private static final String PREFERENCES_KEY_SESSION = "session";
```

Comment:
I see your point, but we should discuss the implementation, as it's not straightforward